#!/usr/bin/env python
PACKAGE="vision_tasks"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("front_clahe_clip", double_t, 0, "CLAHE clip limit", 4.0, 0.0, 40.0);
gen.add("front_clahe_grid_size", int_t, 0, "Bilateral filter on CLAHE iterations", 8, 0, 16);
gen.add("front_clahe_bilateral_iter", int_t, 0, "Bilateral filter on CLAHE iterations", 8, 0, 16);
gen.add("front_balanced_bilateral_iter", int_t, 0, "Bilateral filter on white balanced iterations", 4, 0, 8);
gen.add("front_denoise_h", double_t, 0, "Denoising h value", 10.0, 0, 20.0);
gen.add("front_hough_minline", int_t, 0, "Minimum Length of line", 50, 0, 500);
gen.add("front_hough_threshold", int_t, 0, "Hough threshold", 50, 0, 500);
gen.add("front_hough_maxgap", int_t, 0, "Maximum gap in line", 50, 0, 500);
gen.add("front_canny_threshold_low", int_t, 0, "Lower bound of canny", 50, 0, 500);
gen.add("front_canny_threshold_high", int_t, 0, "Higher bound of canny", 50, 0, 500);
gen.add("front_low_b", int_t, 0, "Lower Bound Of B", 10, 0, 255);
gen.add("front_high_b", int_t, 0, "Higher Bound Of B", 90, 0, 255);
gen.add("front_low_g", int_t, 0, "Lower Bound Of G", 0, 0, 255);
gen.add("front_high_g", int_t, 0, "Higher Bound Of G", 255, 0, 255);
gen.add("front_low_r", int_t, 0, "Lower Bound Of R", 0, 0, 255);
gen.add("front_high_r", int_t, 0, "Higher Bound Of R", 255, 0, 255);
gen.add("front_closing_mat_point", int_t, 0, "", 1, 1, 7);
gen.add("front_closing_iter", int_t, 0, "Closing iterations", 1, 1, 10);

gen.add("bottom_clahe_clip", double_t, 0, "CLAHE clip limit", 4.0, 0.0, 40.0);
gen.add("bottom_clahe_grid_size", int_t, 0, "Bilateral filter on CLAHE iterations", 8, 0, 16);
gen.add("bottom_clahe_bilateral_iter", int_t, 0, "Bilateral filter on CLAHE iterations", 8, 0, 16);
gen.add("bottom_balanced_bilateral_iter", int_t, 0, "Bilateral filter on white balanced iterations", 4, 0, 8);
gen.add("bottom_denoise_h", double_t, 0, "Denoising h value", 10.0, 0, 20.0);
gen.add("bottom_low_b", int_t, 0, "Lower Bound Of B", 10, 0, 255);
gen.add("bottom_high_b", int_t, 0, "Higher Bound Of B", 90, 0, 255);
gen.add("bottom_low_g", int_t, 0, "Lower Bound Of G", 0, 0, 255);
gen.add("bottom_high_g", int_t, 0, "Higher Bound Of G", 255, 0, 255);
gen.add("bottom_low_r", int_t, 0, "Lower Bound Of R", 0, 0, 255);
gen.add("bottom_high_r", int_t, 0, "Higher Bound Of R", 255, 0, 255);
gen.add("bottom_closing_mat_point", int_t, 0, "", 1, 1, 7);
gen.add("bottom_closing_iter", int_t, 0, "Closing iterations", 1, 1, 10);

exit(gen.generate(PACKAGE, "vision_tasks", "gateRange"));
